ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -157,27 +150,73 @@ export const RuntimePanel = withCurrentWorkspace()(({workspace}) => {
                     closeOnClick
                     content={
                       <React.Fragment>
-                        <MenuItem style={styles.presetMenuItem}>General purpose analysis</MenuItem>
-                        <MenuItem style={styles.presetMenuItem}>Genomics analysis</MenuItem>
+                        {
+                          fp.flow(
+                            fp.filter(['displayName', 'General Analysis']),
+                            fp.toPairs,
+                            fp.map(([i, preset]) => {
+                              return <MenuItem
+                              style={styles.presetMenuItem}
+                              key={i}
+                              onClick={() => {
+                                // renaming to avoid shadowing
+                                let presetDiskSize;
```

Comment:
My 2c: I don't want to place a prohibition on `let` (or non-final Java vars) in the codebase - ultimately the metric to optimize for is readability. Using let instead of const reduces readability, since the reader must consider cognitive overhead of potential reassignment. However, this is not the only dimension to consider: for example, it might be a lot more complicated to assign something like this in a massive single statement.

Mostly this is personal preference - I would consider these kinds of suggestions optional unless the proposed alternative is clearly superior. In this particular case, I found the previous version (let, if, else) easier to read - though it is interesting to see how it would be done in lodash fp / functional style and I don't have a strong objection to doing it this way.